### PR TITLE
Test: main page 테스트 코드 구현 

### LINKED ISSUE #69: 메인 패이지 테스트 코드 구현

### PR DESCRIPTION
**Pull request by gahyuun**
### ⛳️ Task

- [X] main page test code 작성
close #69 

### ✍️ Note

**msw 사용해서 api test 했습니다 ! msw를 통해서 setUpServer 설정했는데 저희 프로젝트는 setUpWorker를 통해 브라우저에서 요청을 날리고 화면에 보이는지만 test하면 되기에 엣지 케이스가 아닌 정상 케이스에서 굳이 setUpServer를 할 필요가 있었나 생각이 드네요........ 이 부분은 조금 더 알아보고 추후에 수정하도록 하겠습니다!**

- useNavigate를 사용하면 문제가 있을 수 있는데 rendering 하는 컴포넌트를 BrowserRouter로 감싸주면 됩니다
또한 event를 act 함수에 선언해주세요! ( 이 부분은 error가 아닌 warning입니다)
useNavigate는 Router 안에서 실행 되어야 하는데 저희는 컴포넌트만 렌더링해서 테스트하기에 문제가 생기는 것이고 BrowserRouter로 감싸줘야합니다!

msw 관련 오류를 해결하느라 jest 설정을 변경했는데 설명은 discussions 참고해주세요!
https://github.com/Upjuyanolja/Upjuyanolja_FE/discussions/89

### 📸 Screenshot
image

테스트 실패는 비즈니스 로직 구현을 통해 해결할 예정입니다:)

### 📎 Reference
https://www.fronttigger.dev/2022/react/msw-test
https://kentcdodds.com/blog/stop-mocking-fetch#colocation-and-erroredge-case-testing
https://cheeseb.github.io/testing/react-testing-mocking/#msw%EB%A5%BC-%EC%82%AC%EC%9A%A9%ED%95%9C-%EB%AA%A8%ED%82%B9-%EC%98%88%EC%8B%9C

### CODE REVIEW

File: src/test/main/Main.test.tsx [hunk 1]
```diff
@@ -1,17 +1,58 @@
...
+      </BrowserRouter>,
+    );
+    const navigateButton = screen.getByTestId('navigate-coupon');
+    act(() => {
+      userEvent.click(navigateButton);
+    });
+    expect(window.location.pathname).toBe('/coupon');
+  });
+  test('서버로부터 쿠폰 사용량을 응답 받으면 컴포넌트에 쿠폰 사용량이 출력된다', async () => {
+    server.use(
```
Comment (yangjaehyuk) [inline src/test/main/Main.test.tsx:40]: 아 이렇게 server.use로 msw열어서 쓰는군요 !

**Review by yangjaehyuk — APPROVED** [review]
고생하셨습니다 ~~

**Review by xxxjinn — APPROVED** [review]
수고하셨습니다! 테스트코드 짤 때 참고하겠습니다

**Review by YongYong21 — APPROVED** [review]
수고하셨습니다. 저도 코드 작성할 때 참고해서 잘 작성하도록 하겠습니다.

**Review by syb0127 — APPROVED** [review]
수고하셨습니다! 저도 나중에 잘 참고할게요오~

**Review by tkyoun0421 — APPROVED** [review]
고생 많이 하셨습니다 👍 👍
